pkg/controller/daemon/daemoncontroller.go

```diff
@@ -713,6 +713,16 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *api.Node, ds *exte
 	for _, r := range reasons {
 		glog.V(2).Infof("GeneralPredicates failed on pod %s for reason: %v", newPod.Name, r.GetReason())
 	}
+	if !fit {
+		return false
```

Comment:
In GeneralPredicates we accumulate all the failure reasons, so if we want to do the same thing here I guess we'd drop this if statement and continue with checking PodToleratesNodeTaints regardless of whether fit is already false. But since this is only going into a log message I think it's not worth doing it that way (if it were going into events the user could see, then having the full set of failure reasons could be useful).